collections/collections.go

```diff
@@ -0,0 +1,90 @@
+package collections
+
+import (
+	"errors"
+	"math"
+
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+)
+
+var (
+	// ErrNotFound is returned when the provided key is not present in the StorageProvider.
+	ErrNotFound = errors.New("collections: not found")
+	// ErrEncoding is returned when something fails during key or value encoding/decoding.
+	ErrEncoding = errors.New("collections: encoding error")
+)
+
+// StorageProvider represents sdk.Context
+// it is used to avoid to reduce dependencies.
```

Comment:
```suggestion
// StorageProvider represents sdk.Context and is used to avoid to reduce dependencies.
```

The sentence seems a bit confusing....reduce dependencies?